packages/csharp/PortingAssistant/PortingAssistant.Common/Services/AssesmentService.cs

```diff
@@ -30,25 +33,64 @@ public class AssessmentService : IAssessmentService
         {
             try
             {
+                var startTime = DateTime.Now;
```

Comment:
Could you move all of those metrics collection into one single file and create a single function, pass solutionAlysisReulst as input?